doc/guides/device_mgmt/smp_protocol.rst

```diff
@@ -0,0 +1,1252 @@
+
+    +-----------------------+---------------------------------------------------+
+    | "off"                 | offset to start/continue uplad  at                |
+    +-----------------------+---------------------------------------------------+
+    | "data"                | chunk of data to write to the file;               |
+    |                       | it is CBOR encoded with length embeeded           |
+    +-----------------------+---------------------------------------------------+
+    | "name"                | absolute path to a file                           |
+    +-----------------------+---------------------------------------------------+
+    | "len"                 | length of file, this field is only mandatory      |
+    |                       | when "off" is 0                                   |
+    +-----------------------+---------------------------------------------------+
+
+File upload response
+====================
+
+File upload response header
+
+.. table::
+    :align: center
+
+    +--------+--------------+----------------+
+    | ``OP`` | ``Group ID`` | ``Command ID`` |
+    +========+==============+================+
+    | ``3``  | ``8``        |  ``1``         |
+    +--------+--------------+----------------+
+
+CBOR Payload of request command:
+
+    ```
+    {
+        (str,opt)"off"      : (uint)
+        (str)"rc"           : (int)
+        (str,opt)"rsn"      : (str)
+    }
+    ```
+
+where
+
+.. table::
+    :align: center
+
+    +-----------------------+---------------------------------------------------+
+    | "off"                 | offset of last succesfully written data;          |
+    |                       | appears only when "rc" is 0                       |
+    +-----------------------+---------------------------------------------------+
+    | "rc"                  | :ref:`mcumgr_smp_protocol_status_codes`           |
+    +-----------------------+---------------------------------------------------+
+    | "rsn"                 | :ref:`mcumgr_smp_protocol_status_codes`           |
+    +-----------------------+---------------------------------------------------+
+
+Shell management
+================
+
+Shell management allows to pass commands to shell subsystem with use of SMP
+protocol.
+
+Shell command line execute
+==========================
+
+The command allows to execute command line in a similar way to typing it into
+a shell, but both a request and a response are transported with use of SMP.
+
+Shell command line execute request
+==================================
+
+Execute command request header
+
+.. table::
+    :align: center
+
+    +--------+--------------+----------------+
+    | ``OP`` | ``Group ID`` | ``Command ID`` |
+    +========+==============+================+
+    | ``2``  | ``9``        |  ``0``         |
+    +--------+--------------+----------------+
+
+CBOR payload of request command:
+
+    ```
+    {
+        (str)"argv"     : [
+            (str)<cmd>
```

Comment:
Is it correct that the `cmd` in in the array?

Reply:
The decoding in mcumgr says so: https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/mgmt/mcumgr/lib/cmd/shell_mgmt/src/shell_mgmt.c#L29-L54